src/ansys/mapdl/core/hpc/pyhps.py

```diff
@@ -0,0 +1,373 @@
+
+
+def wait_for_completion(project_api, evaluated=True, failed=False, running=False):
+    eval_status = []
+
+    if evaluated:
+        eval_status.append("evaluated")
+
+    if failed:
+        eval_status.append("evaluated")
+
+    if running:
+        eval_status.append("running")
+
+    logger.debug(f"Waiting on project {proj.id} with criteria: {eval_status}")
+    while not project_api.get_jobs(eval_status=eval_status):
+        time.sleep(2)
+
+
+def create_tmp_file(file_name, content):
+    import tempfile
+    import uuid
+
+    dir_ = tempfile.gettempdir()
+    sub_dir = str(uuid.uuid4())
+
+    tmp_file = os.path.join(dir_, sub_dir, file_name)
+    os.makedirs(os.path.join(dir_, sub_dir))
+
+    with open(tmp_file, "w") as fid:
+        fid.write(content)
+
+    return tmp_file
+
+
+def create_pymapdl_pyhps_job(
+    main_file: str,
+    name: str = None,
+    url: str = None,
+    user: str = None,
+    password: str = None,
+    python: float = None,
+    output_files: Optional[Union[str, list]] = None,
+    shell_file: str = None,
+    requirements_file: str = None,
+    extra_files: Optional[Union[str, list]] = None,
+    config_file: str = None,
+    num_cores: int = None,
+    memory: int = None,
+    disk_space: int = None,
+    exclusive: bool = None,
+    max_execution_time: int = None,
+):
+
+    if python not in [2.7, 3.5, 3.6, 3.7, 3.8, 3.9, 3.10, 3.11, 3.12]:
+        warn("Version of Python might not be supported by the cluster")
+
+    if not os.path.exists(main_file):
+        raise ValueError(f"The python file {main_file} must exists.")
+
+    logger.debug(f"Main python file in: {main_file}")
+
+    if not requirements_file:
+        import pkg_resources
+
+        content = "\n".join(
+            [str(p.as_requirement()) for p in pkg_resources.working_set]
+        )
+        requirements_file = create_tmp_file("requirements.txt", content)
+        logger.debug(f"Requirements file in: {requirements_file}")
+
+    if not shell_file:
+        content = f"""
+echo "Starting"
+
+# Starting venv
+python{python} -m venv .venv
+source .venv/bin/activate
+
+# Installing requirements
+pip install -r {os.path.basename(requirements_file)}
+
+# Running script
+python {os.path.basename(main_file)}
+    """
+
+        shell_file = create_tmp_file("main.sh", content)
+        logger.debug(f"Shell file in: {shell_file}")
+
+    if isinstance(extra_files, str):
+        extra_files = extra_files.split(",")
+    elif extra_files is None:
+        extra_files = []
+
+    if extra_files and not all([os.path.exists(each) for each in extra_files]):
+        raise ValueError("One or more extra files does not exist.")
+
+    input_files = extra_files
+    input_files.append(requirements_file)
+    input_files.append(shell_file)
+    input_files.append(main_file)
+
+    if not output_files:
+        output_files = []
+    elif isinstance(output_files, str):
+        output_files = output_files.split(",")
+
+    # Login
```

Comment:
```suggestion
    # Log in
```
Style guide says "Sign in" is preferred. Log in (two words) as a verb, and one word as a noun.